src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,92 @@
+package duke;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * Deals with loading and saving tasks from file.
+ */
+class Storage {
+
+    /** Variable to store file path for I/O. */
+    private String filePath;
+
+    /**
+     * Constructor used to create file storage.
+     *
+     * @param filePath Filepath of input and output file.
+     */
+    Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads existing file.
+     *
+     * @return A list of list of strings, outer list for many lines in file;
+     * inner list is one line in file, split by delimiter
+     * @throws DukeException When IO error occurs.
+     */
+    ArrayList<ArrayList<String>> load() throws DukeException {
```

Comment:
why is this not called after the constructor instead of calling it in duke.java?